src/Symfony/Component/Notifier/Bridge/SmsSluzba/SmsSluzbaTransport.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\SmsSluzba;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Component\Serializer\Encoder\XmlEncoder;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Dennis Fridrich <fridrich.dennis@gmail.com>
+ */
+final class SmsSluzbaTransport extends AbstractTransport
+{
+    protected const HOST = 'smsgateapi.sms-sluzba.cz';
+
+    private string $username;
+    private string $password;
+
+    public function __construct(
+        #[\SensitiveParameter] string $username,
+        #[\SensitiveParameter] string $password,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null
+    ) {
+        $this->username = $username;
+        $this->password = $password;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('sms-sluzba://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage && (null === $message->getOptions() || $message->getOptions(
+        ) instanceof SmsSluzbaOptions);
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = sprintf(
+            'https://%s/apixml30/receiver?login=%s&password=%s',
```

Comment:
Do they really transport the password via http query?

Reply:
Unfortunately yes. They have actually 3 endpoints you can use, but if you want to get SMS ID back you have to use this one with password in URL. I do not like it as well, this is really old but still much used service. I hope they will update some day.

By the way they return HTTP 200 to all responses even errored ones.